src/main/java/io/confluent/connect/jdbc/JdbcSinkConnector.java

```diff
@@ -73,34 +75,29 @@ public Config validate(Map<String, String> connectorConfigs) {
     /** get configuration parsed and validated individually */
     Config config = super.validate(connectorConfigs);
 
-    configValue(config, PK_MODE)
-        .filter(cfg -> cfg.errorMessages().isEmpty()) // only if individual validation passed.
-        /**
-         * use recommended values for configuration validation
-         * {@link PrimaryKeyModeRecommender#validValues}
-         */
-        .ifPresent(pkMode -> {
-          if (!pkMode.recommendedValues().contains(pkMode.value())) {
-
-            String error = "'" + pkMode.value() + "' is invalid";
-
-            if (configValue(config, DELETE_ENABLED)
-                .map(deleteEnabled -> (Boolean) deleteEnabled.value())
-                .orElse(Boolean.FALSE)) {
-              error += ": Deletes are only supported for pk.mode 'record_key'";
-            }
-
-            pkMode.addErrorMessage(error);
-          }
-        });
+    return validateDeleteEnabledPkMode(config);
+  }
 
+  private Config validateDeleteEnabledPkMode(Config config) {
+    configValue(config, DELETE_ENABLED)
+        .filter(deleteEnabled -> Boolean.TRUE.equals(deleteEnabled.value()))
+        .ifPresent(deleteEnabled -> configValue(config, PK_MODE)
+            .ifPresent(pkMode -> {
+              if (!RECORD_KEY.name().toLowerCase(Locale.ROOT).equals(pkMode.value())) {
```

Comment:
It looks like we allow the `pk.mode` to be either uppercase or lowercase: https://github.com/confluentinc/kafka-connect-jdbc/blob/3de78db9bfada36d575dc82c23aa8abe356fcda3/src/main/java/io/confluent/connect/jdbc/sink/JdbcSinkConfig.java#L581-L582

Would this report an error to the user if they have deletes enabled but specify a `pk.mode` of `RECORD_KEY` instead of `record_key`?

Reply:
that is quick. thank you!

you are right. should accept both RECORD_KEY and record_key.